ledger-tool/src/main.rs

```diff
@@ -1071,10 +1071,16 @@ fn load_bank_forks(
     }
 
     let account_paths = if let Some(account_paths) = arg_matches.value_of("account_paths") {
-        if !blockstore.is_primary_access() {
-            // Be defensive, when default account dir is explicitly specified, it's still possible
-            // to wipe the dir possibly shared by the running validator!
-            eprintln!("Error: custom accounts path is not supported under secondary access");
+        let allow_custom_accounts_path = arg_matches.is_present("allow_custom_accounts_path");
+        if !blockstore.is_primary_access() && !allow_custom_accounts_path {
```

Comment:
@yhchiang-sol - A little context here:

At one point, we had a mechanism that was open-with-primary-and-fall-back-to-secondary. This mechanism more or less told us if the validator process was running as we'd have primary access if it was not, or secondary if it was. We used this to gate whether we should allow the use of a custom accounts path, or whether we should use some alternative path to avoid having the solana-validator and solana-ledger-tool clobber each others' accounts directory.

Then, we changed commands like verify (that do not need to modify the blockstore) to explicitly be secondary mode. A consequence of this check is that a custom accounts path (such as a tmpfs path to use RAM) could no longer be used for verify.

This flag is an attempt to re-allow that behavior; however, I think auto-detecting whether another process has primary would be nicer than forcing use of the flag. As far as I can tell, there is a file called `LOCK` within the rocksdb directory that is used as a filelock. Is this file how we'd determine if another process is in primary? It would definitely seem a little hacky to do this outside of RocksDB, but not sure that there is an API exposed to check whether some other process has primary.

Another option would be attempting to open another session to this same ledger with Primary; if we succeed, that means nobody else had primary and we can continue.

Thoughts? Again, I don't like the flag because it is clunky to have to specify it, but I like having discussion with code in view so figured I'd crack the PR to begin the discussion

Reply:
> This flag is an attempt to re-allow that behavior; however, I think auto-detecting whether another process has primary would be nicer than forcing use of the flag. As far as I can tell, there is a file called LOCK within the rocksdb directory that is used as a filelock.

Pretty much it is.  More specifically, if one process fails to grab the lock for the LOCK file, then it means someone else is running this rocksdb with primary access.

If we are trying to prevent users from shooting their own feet, I am leaning toward having a specific flag.

Or, can I know more about the past issues that are related to this?

Reply:
> Pretty much it is. More specifically, if one process fails to grab the lock for the LOCK file, then it means someone else is running this rocksdb with primary access.

Got it; rocksdb is probably old enough that this wouldn't change, but would still be pretty hacky to directly lock this file so I think we can probably dispel that idea right off the bat.

> If we are trying to prevent users from shooting their own feet, I am leaning toward having a specific flag.

That's fair; the arg comes with an implicit "I know what I'm doing" agreement. However, supposing someone is running through their command history, they might re-run the command without thinking while the validator is now running.

Whereas ledger-tool detecting if someone else has primary access and erroring out in that scenario seems better for usability + avoid a misfire with the new override flag.

> Or, can I know more about the past issues that are related to this?

My understanding is that solana-ledger-tool uses some of the same args as solana-validator, so it follows that you might have a similar command built up for both (same ledger directory / snapshots directory / accounts directory / etc). Because of this and because accounts directory is always a clean start from snapshot, it could be easy for solana-ledger-tool to clobber the accounts directory of validator.

But, validator MUST run with primary access mode.
- The current logic says "allow a custom accounts path if we have primary"
- I want to expand the logic to be "allow a custom accounts path as long as someone else doesn't have primary"